esmvalcore/preprocessor/_area.py

```diff
@@ -355,16 +358,7 @@ def area_statistics(
         agg, agg_kwargs, "cell_area", cube, try_adding_calculated_cell_area
     )
 
-    with warnings.catch_warnings():
-        # Silence various warnings about collapsing multi-dimensional and/or
-        # non contiguous coordinates as this should be fine when the cell areas
-        # are provided and will fail when they needed but not provided.
-        warnings.filterwarnings(
-            "ignore",
-            category=iris.warnings.IrisVagueMetadataWarning,
-            module="iris",
-        )
-        result = cube.collapsed(["latitude", "longitude"], agg, **agg_kwargs)
+    result = cube.collapsed(["latitude", "longitude"], agg, **agg_kwargs)
```

Comment:
How about implementing this as a context manager instead to target only those places in the code where this is relevant, so this could e.g. be written as: 
```python
    with ignore_iris_vague_metadata_warnings:
        result = cube.collapsed(["latitude", "longitude"], agg, **agg_kwargs)
```

Reply:
This was actually my plan at the beginning, but after messing around with https://github.com/ESMValGroup/ESMValCore/pull/2645 too much all I could think about was decorators I guess😄 

Changed in https://github.com/ESMValGroup/ESMValCore/pull/2646/commits/d2b777492a74923b02b3f5d6ed83921e48d698a9.